### PR TITLE
[Feature branch] Fix bug when constructing Spark datasets from file path sources

### PR DESCRIPTION
**Pull request by dbczumar**
## What changes are proposed in this pull request?

Fix bug when constructing Spark datasets from file path sources

## How is this patch tested?

- [ ] Existing unit/integration tests
- [X] New unit/integration tests
- [ ] Manual tests (describe details, including test results, below)

## Does this PR change the documentation?

- [X] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [X] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [X] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [X] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: tests/data/test_spark_dataset.py [hunk 1]
```diff
@@ -12,7 +13,7 @@
 from mlflow.types.utils import _infer_schema
 
 
-@pytest.fixture(autouse=True)
```
Comment (dbczumar) [inline tests/data/test_spark_dataset.py:15]: No reason to autouse the fixture if it isn't required

File: tests/data/test_spark_dataset.py [hunk 2]
```diff
@@ -248,7 +261,7 @@ def test_from_spark_delta_table_name_and_version(spark_session, tmp_path, df):
     # write to delta table
     df_spark.write.format("delta").mode("overwrite").saveAsTable("my_delta_table")
 
-    mlflow_df = mlflow.data.from_spark(df_spark, table_name="my_delta_table", version=1)
```
Comment (dbczumar) [inline tests/data/test_spark_dataset.py:251]: By writing to the table once, it's unclear why `version=1` should work...changing to `version=0`

File: tests/data/test_spark_dataset.py [hunk 3]
```diff
@@ -282,15 +295,19 @@ def test_load_delta_path(spark_session, tmp_path, df):
...
     path = str(tmp_path / "temp.delta")
-    df_spark.write.format("delta").mode("overwrite").save(path)
+
+    df_v0 = pd.DataFrame([[4, 5, 6], [4, 5, 6]], columns=["a", "b", "c"])
+    assert not df_v0.equals(df)
+    df_v0_spark = spark_session.createDataFrame(df_v0)
+    df_v0_spark.write.format("delta").mode("overwrite").save(path)
+
     # write again to create a new version
-    df_spark.write.format("delta").mode("overwrite").save(path)
```
Comment (dbczumar) [inline tests/data/test_spark_dataset.py:289]: The previous test logic wrote the exact same DF twice. On my machine, this didn't create a new version and the test failed. Fixing it here as a drive-by.

File: mlflow/data/spark_delta_utils.py [hunk 4]
```diff
@@ -34,7 +34,7 @@ def _is_delta_table_path(path: str) -> bool:
 
     :return: True if the specified path is a Delta table. False otherwise.
     """
-    if os.path.exists(path) and "_delta_log" in os.listdir(path):
+    if os.path.exists(path) and os.path.isdir(path) and "_delta_log" in os.listdir(path):
```
Comment (dbczumar) [inline mlflow/data/spark_delta_utils.py:37]: The actual fix is just a missing isdir check :)

File: tests/data/test_spark_dataset.py (hunk 3 above)
Comment (prithvikannan) [reply tests/data/test_spark_dataset.py:289]: Thanks for catching and fixing this!

**Review by prithvikannan — APPROVED** [review]
Looks good, thanks @dbczumar for the fix and test improvements

**Comment by dbczumar** [comment]
Test failures are unrelated. Merging :)
